### PR TITLE
DEV: Refactor uppy s3 mulipart mixin to not use self = this

### PR DESCRIPTION
**Pull request by martin-brennan**
We no longer use this pattern, so we can replace with function
binding instead.

### CODE REVIEW

File: app/assets/javascripts/discourse/app/mixins/uppy-s3-multipart.js [hunk 1]
```diff
@@ -18,142 +17,147 @@ export default Mixin.create({
...
       limit: 10,
-      retryDelays,
+      retryDelays: this.retryDelays,
 
-      createMultipartUpload(file) {
-        self._uppyInstance.emit("create-multipart", file.id);
+      createMultipartUpload: this._createMultipartUpload.bind(this),
+      prepareUploadParts: this._prepareUploadParts.bind(this),
+      completeMultipartUpload: this._completeMultipartUpload.bind(this),
+      abortMultipartUpload: this._abortMultipartUpload.bind(this),
```
Comment (CvX) [inline app/assets/javascripts/discourse/app/mixins/uppy-s3-multipart.js:25]: Using `@bind` decorator on methods is preferred instead

Comment (martin-brennan) [reply app/assets/javascripts/discourse/app/mixins/uppy-s3-multipart.js:25]: Huh TIL, nice!

File: app/assets/javascripts/discourse/app/mixins/uppy-s3-multipart.js [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 import { Promise } from "rsvp";
 import { ajax } from "discourse/lib/ajax";
 import AwsS3Multipart from "@uppy/aws-s3-multipart";
 
 export default Mixin.create({
   _useS3MultipartUploads() {
     this.set("usingS3MultipartUploads", true);
-    const self = this;
-    const retryDelays = [0, 1000, 3000, 5000];
+    this.set("retryDelays", [0, 1000, 3000, 5000]);
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/mixins/uppy-s3-multipart.js:10]: is this initialised to null somewhere ?

Comment (CvX) [reply app/assets/javascripts/discourse/app/mixins/uppy-s3-multipart.js:10]: btw. I think that could be a constant

Comment (jjaffeux) [reply app/assets/javascripts/discourse/app/mixins/uppy-s3-multipart.js:10]: yes I don't see this being altered later on ?

File: app/assets/javascripts/discourse/app/mixins/uppy-s3-multipart.js [hunk 3]
```diff
@@ -1,13 +1,13 @@
 import Mixin from "@ember/object/mixin";
+import { bind } from "discourse-common/utils/decorators";
 import { Promise } from "rsvp";
 import { ajax } from "discourse/lib/ajax";
 import AwsS3Multipart from "@uppy/aws-s3-multipart";
+const retryDelays = [0, 1000, 3000, 5000];
```
Comment (CvX) [inline app/assets/javascripts/discourse/app/mixins/uppy-s3-multipart.js:6]:
```suggestion

const RETRY_DELAYS = [0, 1000, 3000, 5000];
```

Comment (martin-brennan) [reply app/assets/javascripts/discourse/app/mixins/uppy-s3-multipart.js]: Done

**Review by CvX — APPROVED** [review]
👌

**Pull request merged**